python/golem_task_api/server.py

```diff
@@ -175,6 +176,22 @@ def __init__(
         await stream.send_message(reply)
         self._shutdown_future.set_result(None)
 
+    @forward_exceptions()
+    async def Shutdown(self, stream):
+        print('Shutdown called')
+        await stream.recv_message()
+        all_tasks = asyncio.Task.all_tasks()
+        print(all_tasks)
+        for task in asyncio.Task.all_tasks():
```

Comment:
Wouldn't that cancel the task that is executing this loop as well?

Reply:
thanks for the heads up! 
tried a local variable first but it did not seem to work, same as last comment, will clean this up when its working in blenderapp